src/components/SearchAndFilter.js

```diff
@@ -47,11 +47,17 @@ class SearchAndFilter extends React.Component {
       <React.Fragment key={length}>
         <label>Ingredient: </label>
         <input type="text" value={ingredients[length]} onChange={(event) => this.setIngredient(length, event)}/>
-        <Button onClick={this.addInput} type="button">+</Button>
+        <button id={`form-add-${length}`} onClick={this.addInput} type="button">+</button>
         <br></br>
       </React.Fragment>
     )
-    this.setState({inputFields: inputs});
+    this.setState({inputFields: inputs}, () => {
+      if (length != 0) {
+        document.getElementById(`form-add-${length-1}`).style.visibility = "hidden";
```

Comment:
prefer to use display:none rather than visibility:none.

display:none will make the whole thing not take up any space, whereas visibility:hidden still has it visible.

Also, sync.

Reply:
done